### ISSUE TITLE
Unit Test Caching caches NGRX functions / data

### ISSUE THREAD

**Issue by rickvandermey** [ngrx/platform#2403]
# 🐞 bug report

Recreated from: https://github.com/angular/angular/issues/35365

After some digging, it seems like all test that fail, has to do something with NGRX. sometimes `actions.pipe`, sometimes data which differ from the expected or initialState.
_______________
After upgrading from Angular 8 to 9 there is an issue while running `ng test`. Because of the randomness of the unit test, the higher amount of test, the more likely some of them fails. When test each component on each stand-alone with `fdescribe` all of them pass.

because we use the tests to validate if the application is still good to deploy, we can't deploy at the moment. 

### Is this a regression?

Yes, the previous version in which this bug was not present was: Angular 8

## 🔬 Minimal Reproduction

https://github.com/rickvandermey/angular-starterkit git clone and run `ng test` multiple times, because this repo only consist of 55 test, its more likely to pass than larger scaled applications.

same commit
Build Pass: https://travis-ci.com/rickvandermey/angular-starterkit/builds/148268061
Build Fail: https://travis-ci.com/rickvandermey/angular-starterkit/builds/148268083

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 9.0.1
Node: 10.15.2
OS: darwin x64

Angular: 9.0.0
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... platform-server, router, service-worker
Ivy Workspace: Yes

Package                                    Version
--------------------------------------------------------------------
@angular-devkit/architect                  0.900.1
@angular-devkit/build-angular              0.900.1
@angular-devkit/build-optimizer            0.900.1
@angular-devkit/build-webpack              0.900.1
@angular-devkit/core                       9.0.1
@angular-devkit/schematics                 9.0.1
@angular/cli                               9.0.1
@ngtools/webpack                           9.0.1
@nguniversal/common                        9.0.0
@nguniversal/express-engine                9.0.0
@nguniversal/module-map-ngfactory-loader   9.0.0-next.9
@schematics/angular                        9.0.1
@schematics/update                         0.900.1
rxjs                                       6.5.4
typescript                                 3.7.5
webpack                                    4.41.2
</code></pre>

NGRX packages
<pre><code>
"@ngrx/effects": "8.6.0",
"@ngrx/entity": "8.6.0",
"@ngrx/router-store": "8.6.0",
"@ngrx/store": "8.6.0",
"@ngrx/store-devtools": "8.6.0",
</code></pre>

update*: 
The same problem occurs when updating to `9.0.0-beta1`

**Comment by rickvandermey** [comment]
Any update on this issue. Is this something the team recognizes as a bug and will be on the roadmap?

**Comment by timdeschryver** [comment]
Isn't this an issue because `actions$` doesn't get assigned in some specs?

```
it('should not dispatch', () => {
			const effect = new RouterEffects(actions$, router, location);
			const metadata = getEffectsMetadata(effect);

			expect(metadata.navigateForward$).toEqual({
				dispatch: false,
				resubscribeOnError: true,
			});
});
```

**Comment by rickvandermey** [comment]
It does seem this brings back the stability for this use case. 
I need to try this out on a bigger project if this will fix the stability on all 1500 unit tests

**Comment by rickvandermey** [comment]
Well, in the end I was able to fix it. The problem is that TestBed gets cache across different spec files.
This means that `provideMockStore({ initialState }),` gets cached aswell. 

We've created a dummyStore which is a complete blueprint of the store and use this for our testcases. but the funny part is, when you update the mockstore with `store.setState()` the output is cached aswell across different spec's. We had to reset / clone the dummyStore at the beginning of each spec file, so we are sure the correct data is used for those cases.

Will close this issue
